docs/quickstart/quickstart_chains/ethereum-uniswap.md

```diff
@@ -0,0 +1,264 @@
+
+```graphql
+type Pool @entity {
+  id: ID!
+  ...
+  swaps: [Swap!]! @derivedFrom(field: "pool")
+  ...
+}
+
+type Swap @entity {
+  id: ID!
+  ...
+  pool: Pool!
+  ...
+}
+```
+
+::: warning Important
+When you make any changes to the schema file, please ensure that you regenerate your types directory.
+:::
+
+SubQuery makes it easy and type-safe to work with your GraphQL entities, as well as smart contracts, events, transactions, and logs. SubQuery CLI will generate types from your project's GraphQL schema and any contract ABIs included in the data sources.
+
+::: code-tabs
+@tab:active yarn
+
+```shell
+yarn codegen
+```
+
+@tab npm
+
+```shell
+npm run-script codegen
+```
+
+:::
+
+This will create a new directory (or update the existing) `src/types` which contain generated entity classes for each type you have defined previously in `schema.graphql`. These classes provide type-safe entity loading, read and write access to entity fields - see more about this process in [the GraphQL Schema](../../build/graphql.md). All entites can be imported from the following directory:
+
+```ts
+// Import entity types generated from the GraphQL schema
+import { Bundle, Burn, Factory, Mint, Pool, Swap, Tick, Token } from "../types";
+```
+
+As you're creating a new Etheruem based project, this command will also generate ABI types and save them into `src/types` using the `npx typechain --target=ethers-v5` command, allowing you to bind these contracts to specific addresses in the mappings and call read-only contract methods against the block being processed.
+
+It will also generate a class for every contract event to provide easy access to event parameters, as well as the block and transaction the event originated from. Read about how this is done in [EVM Codegen from ABIs](../../build/introduction.md#evm-codegen-from-abis).
+
+All of these types are written to `src/typs/abi-interfaces` and `src/typs/contracts` directories. In the example Gravatar SubQuery project, you would import these types like so.
+
+```ts
+// Import event types from the registry contract ABI
+import {
+  InitializeEvent,
+  MintEvent,
+  BurnEvent,
+  SwapEvent,
+  FlashEvent,
+} from "../types/contracts/Pool";
+```
+
+Check out the [GraphQL Schema](../../build/graphql.md) documentation to get in-depth information on `schema.graphql` file.
+
+## 3. Add a Mapping Function
+
+Mapping functions define how chain data is transformed into the optimised GraphQL entities that we previously defined in the `schema.graphql` file.
+
+They function in a manner akin to SubGraphs, and you can observe in this Uniswap project that they are distributed across three distinct files, along with the inclusion of various utility files.
+
+Check out our [Mappings](../../build/mapping/ethereum.md) documentation to get more information on mapping functions.
+
```

Comment:
Also, no examples of mapping functions? We could also add some explanation of what is happening in each function or at least the one we showed as an example in the shell.